pkg/reconciler/contour/contour.go

```diff
@@ -189,6 +190,12 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ing *v1alpha1.Ingress) r
 			if port.Name == networking.ServicePortNameH2C {
 				serviceToProtocol[name] = "h2c"
 				break
+
+			} else if cfg.Network != nil && cfg.Network.InternalEncryption {
+				// TODO(KauzClay): Every port should fall into here, except those that are h2c. This would mean that h2c would be unencrypted even if internal encryption is on. Is that okay?
```

Comment:
Technically `h2` requires the connection to be `tls` - otherwise it's `h2c`

If a user specifies `h2c` & internal encryption is on we set probably should set it `h2`.  It'd be nice if that worked :)

Otherwise we'd want `tls` to the queue proxy and then the queue proxy to the user container is `h2c`.

Reply:
I ended up just switching to h2, and things seem to work. I manually deploy the h2 test app and it seemed to work out okay.